tests/common/src/tools/thread_test.cpp

```diff
@@ -140,6 +140,80 @@ TEST(CommonTest, CondVar)
     EXPECT_TRUE(worker.Join().IsNone());
 }
 
+TEST(CommonTest, CondVarTimeout)
```

Comment:
Probably it would better separate them on different test cases.
And I'm not sure we should put all the tests in CommonTest, I would make a separate ThreadTest

Reply:
Put each common class in its own test.